admin/add_copyright.py

```diff
@@ -106,36 +105,62 @@ def addCopyrightToPythonFile(filename, lines):
     return newLines
 
 
+def updateCopyrightString(filename, lines):
+    newLines = list([])
+
+    iterLines = iter(lines)
+    copyrightReached = False
+    for line in iterLines:
+        newLines.append(line)
+        if line == "COPYRIGHT = (":
+            copyrightReached = True
+            break
+    if not copyrightReached:
+        raise RuntimeError(f"Start of copyright not found in {filename}.")
```

Comment:
Explicit state variable is not necessary when using for–else, but this is a matter of taste:

```suggestion
    for line in iterLines:
        newLines.append(line)
        if line == "COPYRIGHT = (":
            break
    else:
        raise RuntimeError(f"Start of copyright not found in {filename}.")
```

Similar for `copyrightEndReached` below.

Reply:
That construct is new for me. I'll change it.